streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -16,23 +16,19 @@
  */
 package org.apache.kafka.streams.kstream.internals;
 
-import org.apache.kafka.streams.processor.AbstractProcessor;
 import org.apache.kafka.streams.processor.Processor;
-import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.ProcessorSupplier;
-import org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl;
-import org.apache.kafka.streams.state.KeyValueStore;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Objects;
 
-public class KTableSource<K, V> implements ProcessorSupplier<K, V> {
-    private static final Logger LOG = LoggerFactory.getLogger(KTableSource.class);
+public abstract class KTableSource<K, V> implements ProcessorSupplier<K, V> {
+    protected static final Logger LOG = LoggerFactory.getLogger(KTableSource.class);
```

Comment:
a logger should always be private -- if classed extend `KTableSource` the should create their own logger with the corresponding "child" class name.